jwql/instrument_monitors/common_monitors/dark_monitor.py

```diff
@@ -0,0 +1,955 @@
+            Tuple of two lists, containing x,y coordinates of bad pixels.
+            (Output of np.where call)
+
+        pixel_type : str
+            Type of bad pixel. Options are 'dead', 'hot' and 'noisy'
+
+        files : list
+            List of fits files which were used to identify the bad pixels
+
+        mean_filename : str
+            Name of fits file containing the mean dark rate image used to
+            find these bad pixels
+
+        baseline_filename : str
+            Name of fits file containing the baseline dark rate image used
+            to find these bad pixels
+        """
+        #logging.info('Adding {} {} pixels to database.'.format(len(coordinates[0]), pixel_type))
+
+        entry = {'detector': self.detector, 'x_coord': coordinates[0], 'y_coord': coordinates[1],
+                 'type': pixel_type, 'source_files': files, 'mean_dark_image_file': mean_filename,
+                 'baseline_file': baseline_filename, 'entry_date': datetime.datetime.now()}
+        self.pixel_table.__table__.insert().execute(entry)
+
+    def get_metadata(self, filename):
+        """Collect basic metadata from a fits file
+
+        Parameters
+        ----------
+        filename : str
+            Name of fits file to examine
+        """
+        header = fits.getheader(filename)
+        try:
+            self.detector = header['DETECTOR']
+            self.x0 = header['SUBSTRT1']
+            self.y0 = header['SUBSTRT2']
+            self.xsize = header['SUBSIZE1']
+            self.ysize = header['SUBSIZE2']
+            self.sample_time = header['TSAMPLE']
+            self.frame_time = header['TFRAME']
+            self.read_pattern = header['READPATT']
+        except KeyError as e:
+            print(e)
+
+    def exclude_existing_badpix(self, badpix, pixel_type):
+        """Given a set of coordinates of bad pixels, determine which of
+        these pixels have been previously identified and remove them
+        from the list
+
+        Parameters
+        ----------
+        badpix : tup
+            Tuple of lists containing x and y pixel coordinates. (Output
+            of numpy.where call)
+
+        pixel_type : str
+            Type of bad pixel being examined. Options are ``hot``, ``dead``,
+            and ``noisy``
+
+        Returns
+        -------
+        new_pixels_x : list
+            List of x coordinates of new bad pixels
+
+        new_pixels_y : list
+            List of y coordinates of new bad pixels
+
+        """
+        if pixel_type not in ['hot', 'dead', 'noisy']:
+            raise ValueError('Unrecognized bad pixel type: {}'.format(pixel_type))
+
+        db_entries = session.query(self.pixel_table) \
+            .filter(self.pixel_table.type == pixel_type) \
+            .filter(self.pixel_table.detector == self.detector) \
+            .all()
+
+        already_found = []
+        if len(db_entries) != 0:
+            for _row in db_entries:
+                print('row:', _row)
```

Comment:
@bhilbert4 Since this will nominally be run in a `cron` job, we could probably remove these `print` statements

Reply:
Done.